Lesson17/baljan/contactBook.js

```diff
@@ -0,0 +1,201 @@
+}
+
+/*
+-----------------------------------------------------------
+  STEP 4: View a Contact by Name
+-----------------------------------------------------------
+Function: viewContact(name)
+- Loops over 'contacts' to find one matching 'name'.
+- Logs the contact info if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+function viewContact(name, contactList) {
+  const contact = findContact(name, contactList);
+
+  if (contact) {
+    console.log(`Name: ${contact.name}, Phone: ${contact.phone}, Email: ${contact.email}`);
+  } else {
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+
+/*
+-----------------------------------------------------------
+  STEP 5: Update a Contact
+-----------------------------------------------------------
+Function: updateContact(name, newPhone, newEmail)
+- Finds the contact by name and updates phone + email.
+- Logs "Contact updated successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+
+function updateContact(name, newPhone, newEmail, contactList) {
+  const isContactFound = findContact(name, contactList);
+
+  if (isContactFound) {
+        contact.phone = newPhone;
+        contact.email = newEmail;
+        console.log(`Contact updated: ${name}`);
+        return;
+      } else {
+    console.log(`No contact found with the name: ${name}`);
+  }
+
+}
+
+/*
+-----------------------------------------------------------
+  STEP 6: Remove a Contact
+-----------------------------------------------------------
+Function: removeContact(name)
+- Finds the index of the contact with 'name' using
+  findIndex() or a loop.
+- Splices it from the array if found.
+- Logs "Contact removed successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+function removeContact(name, contactList) {
+  const index = contactList.findIndex(contact => contact.name === name);
+
+  if (index !== -1) {
+    contactList.splice(index, 1);
+      console.log("Contact removed successfully.");
+      return;
+    } else {
+  console.log(`No contact was found with name: ${name}`);
+}
+}
+
+
+
+/*
+-----------------------------------------------------------
+  STEP 7: Testing Our Functions
+-----------------------------------------------------------
+Below are some sample function calls to demonstrate the
+Contact Book in action.
+*/
+
+console.log("Initial contact list:");
+displayAllContacts(contacts);
+
+console.log("\nAdding a new contact: Charlie");
+addContact("Charlie", "777-777-7777", "charlie@example.com", contacts);
+displayAllContacts(contacts);
+
+console.log("\nAdding a new contact: Charlie");
+addContact("Charlie", "777-777-7777", "charlie@example.com", contacts);
+displayAllContacts(contacts);
+
+console.log("\nViewing Charlie's contact:");
+viewContact("Charlie", contacts);
+
+
+console.log("\nUpdating Charlie's contact:");
+updateContact("Charlie", "999-999-9999", "charlie@updated.com");
```

Comment:
You need to pass contacts as the last parameter to this function as well.
```suggestion
updateContact("Charlie", "999-999-9999", "charlie@updated.com", contacts);
```